### PR TITLE
Update ESLint config from v4 to v8

### PR DESCRIPTION
**Pull request by Gudahtt**
The ESLint config has been updated a few major versions. The most notable additions are the splitting of the `@metamask/eslint-config` package, `prettier`, the `import` plugin, the `packagejson` Prettier plugin, and semicolons.

The lint npm scripts were updated to match the MetaMask module template.

All lint violates were fixed using `yarn lint:fix`.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,24 +1,18 @@
...
-  extends: [
-    '@metamask/eslint-config',
-    '@metamask/eslint-config/config/mocha',
-    '@metamask/eslint-config/config/nodejs',
-  ],
+  extends: ['@metamask/eslint-config', '@metamask/eslint-config-nodejs'],
 
   parserOptions: {
     ecmaVersion: 2017,
   },
```
Comment (rekmarks) [inline .eslintrc.js:8]: This can be omitted: https://github.com/MetaMask/eslint-config/blob/main/packages/base/src/index.js#L8-L19

**Review by rekmarks — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (Gudahtt) [reply .eslintrc.js:8]: Good catch!

Comment (Gudahtt) [reply .eslintrc.js]: Fixed in https://github.com/MetaMask/eth-hd-keyring/pull/50/commits/bfd7aa256f28583f2177cb6830f078464971868b

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**